### ISSUE TITLE
Test Automation for All available solutions on the new dashboard

### ISSUE THREAD

**Issue by ramezsaeed** [threefoldtech/tfgrid-sdk-ts#1841]
Add automated tests for all solutions in dashboard.
- Tests must include ssh to the created solution.
- successful login if the solution has UI interface

**Comment by khaledyoussef24** [comment]
Wip
- added a new solutions page to the selenium 
- Added a separate page for solutions (full vm and micro vm)
- On going automated test for deploying still working on verifying the deployment and asserting then will go to deleting.

**Comment by xmonader** [comment]
it should be scripted first (and added to integration tests)

**Comment by khaledyoussef24** [comment]
WIP 
started in two automated tests but stuck in making them passing
(peertube , funk whale)

**Comment by khaledyoussef24** [comment]
work done:
- added scripts to all of the 12 soltuions 
- domain generating still needs some fixing (tested on public ips)
- finished 5 solutions but could not test them freely due to lack of public ips and the planetary ip current issue and development to use mycellium.

WIP:
finishing the rest of the tests.
adding to ping the solutions through ip with time out to assert that the websites made.
tests left to be done

- [ ] owncloud
- [ ] presearch
- [ ] subsquid
- [ ] casperlabs
- [ ] node pilot
- [ ] umbrel
- [ ] wordpress

**Comment by khaledyoussef24** [comment]
the new branch is ready for review 
solved all the conflicts

**Comment by khaledyoussef24** [comment]
work done :
- added a http request to the test to check if the gateway deployed right.
- added mycelium flag to the tests and scripts.
- the pr is ready for review.

**Comment by khaledyoussef24** [comment]
tried scripts and working fine.
tests don't work with consistency, but still work.
 PR is ready for review.

**Comment by khaledyoussef24** [comment]
some changes are made so scripts needs to be updated.
comments are resolved but still searching why scripts are not working.

**Comment by khaledyoussef24** [comment]
Work done
scripts for solutions are added.
the vms are working fine with flists waiting to be tested with public ips then will move the pr from drafts.
(created a new branch because the old branch had so many conflicts.)
will continue the tests after the pr gets merged.

**Comment by khaledyoussef24** [comment]
estimated time for the tests 3 days after the test run of testnet andthe script branch gets merged.

**Comment by A-Harby** [comment]
https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2743

**Comment by A-Harby** [comment]
> #2743

This PR got accepted and merged, and I created another PR with remaining solutions scripts here https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2766.
And now pending reviews.

**Comment by A-Harby** [comment]
I created the a third PR for the automated tests for the casperlabs and waiting for the reviews. https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2784

**Comment by A-Harby** [comment]
Updated PR with Peertube, Funkwhale, and CasperLabs, and still waiting for review.

And got blocked on https://github.com/threefoldtech/tf-images/issues/250 and https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2833.

**Comment by A-Harby** [comment]
As 3 automated tests scripts are megered https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2784, I created another PR with the remaining 8 applications https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890

**Comment by A-Harby** [comment]
Created a separate PR for Algorand and presearch: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3030.

**Comment by A-Harby** [comment]
By having last [PR](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890#event-13944781590) merged this issue should be done.

All of the 13 scripts for the applications are created in example folder, And their tests are created with Jest on tests folder.
